src/filemap.rs

```diff
@@ -85,21 +85,38 @@ pub fn write_file(text: &StringBuffer,
         }
     }
 
+    fn diff(text: &StringBuffer,
+            filename: &str,
+            config: &Config)
+            -> Result<Vec<Mismatch>, io::Error> {
+        let mut f = try!(File::open(filename));
+        let mut ori_text = String::new();
+        try!(f.read_to_string(&mut ori_text));
+        let mut v = Vec::new();
+        try!(write_system_newlines(&mut v, text, config));
```

Comment:
For the backup check you should probably stop after reading in the new file and then just do an == comparison rather than generating the diff, which seems wasteful. Also, you probably want to compare with the newlines we would actually output so we make a backup in the case that we only change newlines